File: 2022/03/06/arrangingCoins.ts
/**
 * https://leetcode.com/problems/arranging-coins/
 * @author Andrew Jarombek
 * @since 3/6/2022
 */

function arrangeCoins(n: number): number {
    let rows = 0;

    while (n > 0) {
        if (n >= rows + 1) {
            rows += 1;
        }

        n -= rows;
    }

    return rows;
}

File: 2023/06/24/go/arranging_coins.go
/**
 * https://leetcode.com/problems/add-strings/
 * Author: Andrew Jarombek
 * Date: 6/24/2023
 */

package _go

func arrangeCoins(n int) int {
	steps := 0

	for n > steps {
		steps++
		n -= steps
	}

	return steps
}
